### PR TITLE
Refactored the rake tasks

### PR DESCRIPTION
**Pull request by AmartC**

### CODE REVIEW

File: lib/tasks/methods.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
```
[the remark this answers is not in the document]
Comment (rdingwell) [reply lib/tasks/methods.rb]: Please change all of the methods to select_* instead of obtain_*

File: lib/tasks/hdm.rake [hunk 2]
```diff
@@ -50,48 +51,14 @@ namespace :hdm do
...
-    if profiles.empty?
-      puts 'User has no profile, please create a profile before continuing'
-      exit(0)
-    elsif profiles.length > 1
-      puts 'Select which profile you want to load the data into '
-      profiles.each_with_index { |p, i| puts "#{i}. #{p.name}" }
-      profile_indx = STDIN.gets.chomp.to_i
-    else
-      puts "User only has a single profile, using profile #{profiles[profile_indx.to_i].name}"
+      user_indx = obtain_user_index
```
Comment (rdingwell) [inline lib/tasks/hdm.rake]: see comment in the methods file about what to return here

File: lib/tasks/provider_client.rake [hunk 3]
```diff
@@ -2,57 +2,19 @@
...
-      puts 'Select which provider you want to associate the EDR with:'
-      providers.each_with_index { |p, i| puts "#{i} #{p.name}" }
-      puts '(Note: other providers may exist but do not have applications setup.'
-      puts ' Run the hdm:create_provider_application rake task to create the application)'
-      provider_indx = STDIN.gets.chomp.to_i
-    else
-      puts "There is only 1 provider with an application in the system, using provider #{providers[provider_indx.to_i].name}"
-    end
-    provider = providers[provider_indx.to_i]
+    user_indx = obtain_user_index
```
Comment (rdingwell) [inline lib/tasks/provider_client.rake]: see comment in methods file about what to return here

File: lib/tasks/provider_client.rake [hunk 4]
```diff
@@ -2,57 +2,19 @@
 
 require 'rake'
 
-namespace :provider_client do
+namespace :provider_client_cp do
```
[the remark this answers is not in the document]
Comment (AmartC) [reply lib/tasks/provider_client.rake]: oops that was a line i was using for testing. i'll get rid of it.

File: lib/tasks/hdm.rake [hunk 5]
```diff
@@ -22,7 +24,7 @@ namespace :hdm do
   task :load, %i[file create_user account_email account_password] => :environment do |_t, args|
     bundle_json = File.open(args.file, 'r:UTF-8').read
     json_format = JSON.parse(bundle_json)
-    user_indx = 0
+    user = ''
     if args.create_user == 'y'
       patient_entry = json_format['entry'].find \
```
Comment (rdingwell) [inline lib/tasks/hdm.rake]: I didn't notice this before but you should also move the functionality of creating  the user from a fhir bundle into the  library file as well

**Pull request merged**